### PR TITLE
fix(vsc): create project experiment blocked issue

### PR DESCRIPTION
**Pull request by tecton**
@microsoft/teamsfx-cli	v1.0.4-rc.1
ms-teams-vscode-extension	v4.0.4-rc.1

CLI BREAKING CHANGE:
  

Extension-toolkit BREAKING CHANGE:
 

CLI feat commits:
  

Extension-toolkit feat commits:

### CODE REVIEW

**Comment by tecton** [comment]
Related bug: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/15180092

File: packages/vscode-extension/src/handlers.ts [hunk 1]
```diff
@@ -468,21 +468,19 @@ export async function openFolder(
...
+      VS_CODE_UI.showMessage(
+        "info",
+        localize("teamstoolkit.handlers.openProject.title"),
+        false,
+        localize("teamstoolkit.handlers.openInNewWindow"),
+        localize("teamstoolkit.handlers.openInCurrentWindow")
+      ),
+    ]);
     if (selection.isOk()) {
-      clearTimeout(autoOpenTimeout);
```
Comment (HuihuiWu-Microsoft) [inline packages/vscode-extension/src/handlers.ts:485]: The 'vscode.openFoler' command will be executed in 2s. Will it time out to open in new window during this time if don't clear time out here?

Comment (tecton) [reply packages/vscode-extension/src/handlers.ts:485]: This PR uses `Promise.race()` instead of setTimeout&clearTimeout approach. It can be awaited.

**Review by HuihuiWu-Microsoft — APPROVED** [review]

**Comment by tecton** [comment]
E2E TEST: N/A

**Pull request merged**